### DISCUSSION TITLE
Problema Caddy e connessione SSH

### DISCUSSION THREAD

**Discussion by fabioperencin** [indomus/forum#1583] [HUB: Home Assistant]
Ciao a tutti,

sto seguendo la vostra guida per migrare da HomeAssistant Core Venv a Docker. Però ho un problema con il docker Caddy. Appena lo inizializzo non riesco più ad accedere in SSH. Cosa sbaglio? Il docker lo inizializzo con le porte 443:443 e 80:80 come da guida.

**Comment by snakuzzo** [comment]
Ciao. Perdonami eh, ma uno come fa a darti una mano con così pochi dettagli? Senza un log, la config, niente di niente.
Poi si dà per scontato di fare tutto in maniera giusta dicendo "ho fatto tutto come da guida", ma la maggior parte delle volte poi non è così.
Aiutaci ad aiutarti...non possiamo prevedere le cose con la sfera di cristallo ;)

**Reply by fabioperencin** [reply]
Ciao, ti aggiungo un pò di dettagli.
- ho instanziato Home assistant Core con docker e funziona.
-  per la parte duckDNS ho tenuto qlo stesso dns che avevo creato per dehydrated, l'aggiornamento viene fatto tramite fritzBox. Il token ho tenuto lo stesso.
- per quanto riguarda il port forwarding il fritzBox è configurato e aperto per il 443
- ho creato le cartelle mkdir -p /home/pi/caddy/data
- ho creato i file Caddyfile con la seguente conf:
```server 
 { 
     acme_ca https://acme-staging-v02.api.letsencrypt.org/directory 
 }
 https://xxx.duckdns.org { 
   respond "Saluti da Fabio" 
   file_server
 }
 ```
dopodiche istanzio caddy  con 
```
docker run --init -d --name="caddy" -e "TZ=Europe/Rome" -v /home/pi/caddy/Caddyfile:/etc/caddy/Caddyfile -v /home/pi/caddy/data:/data --restart always -p 443:443  caddy:latest
```

![image](https://user-images.githubusercontent.com/22098587/159095954-9ca9c64c-2a9e-4afc-b191-f94bf59f4fe3.png)

E a questo punto cade la connessione ssh da putty. Se mi collego direttamente al raspi l'istanza è su e appena la rimuovo torna accessibile da ssh. 
Se hai bisogno di qualcosa in particolare fammi sapere. 
Grazie mille

**Reply by snakuzzo** [reply]
Hai usato il markdown in maniera errata. Cortesemente correggi che non si capisce una fava a livello di indentazione.
Poi...la conf del server di staging ti serve solo per testare che il giro funzioni...ma dopo per ottenere il certificato la devi commentare e restartare il container.
Ma poi tu lì nel Caddyfile non hai ancora configurato il tutto per HA
Resta il fatto che con questi dettagli che hai dato non si può capire il problema.
Non ha senso che ti caschi la connessione SSH in locale...non c'entra niente con Caddy.

**Reply by fabioperencin** [reply]
Scusa ho sistemato i markdown. Si quella era la configurazione di test ma gia li ho avuto questo problema che mi turba non poco. 

> Non ha senso che ti caschi la connessione SSH in locale...non c'entra niente con Caddy.

Eppure appena attivo l'istanza mi isola l'RPI, appena la stoppo torna accessibile tramite ssh. Anche io lo trovo strano ma non riesco a capire quale sia la configurazione che crea questo disagio.

**Reply by snakuzzo** [reply]
mi spiace ma così non è possibile aiutarti. Non ha alcun senso.

**Reply by fabioperencin** [reply]
Eh immagino. Stasera ho distrutto il docker e le cartelle certificati. Creato il composer, istanziato e mi sembra tutto ok ma il rasberry si isola dalla rete… anche i ping a google vanno in timeout.
Stessa cosa se eseguo un traceroute anche verso il fritzbox.

questo il log 
![1D92A722-0116-42CD-BF24-FBD29AF43B7D](https://user-images.githubusercontent.com/22098587/159593197-1e229d70-01cd-479a-8eca-92331454991a.jpeg)



Ho notato anche che quando istanzio caddy crea una interfaccia di rete:
![7F4EC665-843F-4B5B-BDA7-03B8956EB564](https://user-images.githubusercontent.com/22098587/159593263-5fbc6716-d0c8-4f68-985c-a8e4a5889367.jpeg)

Così è Tutto normale?

**Reply by snakuzzo** [reply]
Rimuovo le immagini.
Posta i log in modo umano...utilizzando correttamente il markdown, per cortesia.

**Reply by fabioperencin** [reply]
Eh non entrando in SSH in quanto isolato non riesco a copiarli…

**Reply by snakuzzo** [reply]
allora ok... riposta le foto

**Reply by fabioperencin** [reply]
confermo che non è caddy il problema, ho provato con swag e appena lo avvio il problema è lo stesso... quindi è qualche configurazione di docker o del raspberry... il problema è capire quale sia il problema.. ne sto uscendo pazzo...

**Comment by fabioperencin** [comment]
Buonasera, posto la risoluzione al mio problema dopo aver indagato parecchio. In sostanza utilizzando il driver Bridge, docker crea giustamente una rete virtuale veth*. Per qualche motivo, che ignoro, il dhcpcd assegnava l'ip di questa interfaccia lo stesso ip della wlan0 che incasinava la kernel route rendendo il raspberry fuori dalla rete. 
Per risolvere tale problema ho inserito nel file `/etc/dhcpcd.conf` la riga:

`denyinterfaces veth*`

Riavviato il raspberry. A questo punto l'interfaccia veth viene ignorata dal dhcpcd... Ora tutto funziona perfettamente.

**Reply by akela-it** [reply]
Grazie Fabio era proprio quello che mi serviva! 
Anche io utilizzo raspberry con wifi e avevo lo stesso problema, con la tua indicazione ho risolto.
